numpy/core/src/multiarray/item_selection.c

```diff
@@ -1749,6 +1750,275 @@ PyArray_LexSort(PyObject *sort_keys, int axis)
+        /* TODO: can we deprecate this? */
+    }
+    else if (check_and_adjust_axis(&axis, nd) < 0) {
+        goto fail;
+    }
+    if ((nd == 0) || (PyArray_SIZE(mps[0]) <= 1)) {
+        /* empty/single element case */
+        // Nothing to do here just return
+        goto finish;
+    }
+
+    for (i = 0; i < n; i++) {
+        its[i] = (PyArrayIterObject *)PyArray_IterAllButAxis(
+                (PyObject *)mps[i], &axis);
+        if (its[i] == NULL) {
+            goto fail;
+        }
+    }
+    /* Now do the sorting */
+    sit = (PyArrayIterObject *)
+            PyArray_IterAllButAxis((PyObject *)self, &axis);
+    if (sit == NULL) {
+        goto fail;
+    }
+    if (!object) {
+        NPY_BEGIN_THREADS;
+    }
+    size = sit->size;
+    N = PyArray_DIMS(mps[0])[axis];
+    selstride = PyArray_STRIDE(self, axis);
+    aelsize = PyArray_DESCR(self)->elsize;
+    maxelsize = PyArray_DESCR(mps[0])->elsize;
+    needcopy = (selstride != aelsize);
+    for (j = 0; j < n; j++) {
+        needcopy = needcopy
+            || PyArray_ISBYTESWAPPED(mps[j])
+            || !(PyArray_FLAGS(mps[j]) & NPY_ARRAY_ALIGNED)
+            || (PyArray_STRIDES(mps[j])[axis] != (npy_intp)PyArray_DESCR(mps[j])->elsize);
+        if (PyArray_DESCR(mps[j])->elsize > maxelsize) {
+            maxelsize = PyArray_DESCR(mps[j])->elsize;
+        }
+    }
+    assert(N > 0);  /* Guaranteed and assumed by indbuffer */
+    char *indbuffer = PyDataMem_NEW(N * sizeof(npy_intp));
+    if (indbuffer == NULL) {
+       goto fail;
+    }
+    if (needcopy) {
+        char *valbuffer;
+        int *swaps;
+        npy_intp valbufsize = N * maxelsize;
+        if (NPY_UNLIKELY(valbufsize) == 0) {
+            valbufsize = 1;  /* Ensure allocation is not empty */
+        }
+
+        valbuffer = PyDataMem_NEW(valbufsize);
+        if (valbuffer == NULL) {
+            PyDataMem_FREE(indbuffer);
+            goto fail;
+        }
+        swaps = malloc(NPY_LIKELY(n > 0) ? n * sizeof(int) : 1);
+        if (swaps == NULL) {
+            PyDataMem_FREE(valbuffer);
+            PyDataMem_FREE(indbuffer);
+            goto fail;
+        }
+
+        for (j = 0; j < n; j++) {
+            swaps[j] = PyArray_ISBYTESWAPPED(mps[j]);
+        }
+        while (size--) {
+            iptr = (npy_intp *)indbuffer;
+            for (i = 0; i < N; i++) {
+                *iptr++ = i;
+            }
+            for (j = 0; j < n; j++) {
+                int rcode;
+                elsize = PyArray_DESCR(mps[j])->elsize;
+                astride = PyArray_STRIDES(mps[j])[axis];
+                argsort = PyArray_DESCR(mps[j])->f->argsort[NPY_STABLESORT];
+                if(argsort == NULL) {
+                    argsort = npy_atimsort;
+                }
+                _unaligned_strided_byte_copy(valbuffer, (npy_intp) elsize,
+                                             its[j]->dataptr, astride, N, elsize);
+                if (swaps[j]) {
```

Comment:
`PyArray_ISBYTESWAPPED` is just a few pointer dereferences, most of which you've already made in this loop body anyway. Let's remove this `swaps` array.
```suggestion
                if (PyArray_ISBYTESWAPPED(mps[j])) {
```